libclc/clc/lib/generic/math/clc_sinpi.inc

```diff
@@ -0,0 +1,114 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#if __CLC_FPSIZE == 32
+
+_CLC_OVERLOAD _CLC_DEF __CLC_GENTYPE __clc_sinpi(__CLC_GENTYPE x) {
+  __CLC_INTN ix = __CLC_AS_INTN(x);
+  __CLC_INTN xsgn = ix & (__CLC_INTN)0x80000000;
+  ix ^= xsgn;
+  __CLC_GENTYPE ax = __CLC_AS_GENTYPE(ix);
+  __CLC_INTN iax = __CLC_CONVERT_INTN(ax);
+  __CLC_GENTYPE r = ax - __CLC_CONVERT_GENTYPE(iax);
+  __CLC_INTN xodd =
+      xsgn ^ ((iax & 0x1) != 0 ? (__CLC_INTN)0x80000000 : (__CLC_INTN)0);
```

Comment:
This is another case that can be cleaned up with fabs

Reply:
Ah yes. I actually did a bit of that on `clc_tanpi.inc` this morning but forgot to go back to sin/cos. I'll go back now and do that, but there's definitely more to do across the board.